### PR TITLE
675 glossary form

### PR DESCRIPTION
**Pull request by emmapeel2**
Hello nijel, here I scratched a solution to inspire you, related to my feature request 
https://github.com/nijel/weblate/issues/675

Currently is working on my setup, although I think I maybe added too many fancy crispy objects for just one form...

Please review it, thanks

### CODE REVIEW

File: weblate/trans/views/edit.py [hunk 1]
```diff
@@ -469,6 +470,22 @@ def translate(request, project, subproject, lang):
...
+        addwordform = TinyWordForm(request.POST)
+        if addwordform.is_valid():
+            Dictionary.objects.create(
+                request,
+                project=project,
+                language=lang,
+                source=addwordform.cleaned_data['source'],
+                target=addwordform.cleaned_data['target']
+            )
+            return HttpResponseRedirect(this_unit_url)
```
Comment (nijel) [inline weblate/trans/views/edit.py]: undefined name 'this_unit_url'

**Comment by nijel** [comment]
...also there are some coding style issues (see https://travis-ci.org/nijel/weblate/jobs/66990551#L782)

**Comment by emmapeel2** [comment]
ei nijel thanks for your patience. i have fixed the sloppy code errors. should install pylint over here (blush)

regarding the 'Not enough public methods', maybe we can fuse TinyWordform with WordForm and then it will be less code and cleaner...

File: weblate/trans/views/edit.py (hunk 1 above)
Comment (emmapeel2) [inline weblate/trans/views/edit.py]: Ooops! I shouldn't have moved the code around last minute!

**Comment by emmapeel2** [comment]
Ok nijel I don't think I know how to solve https://travis-ci.org/nijel/weblate/jobs/67012970 and will try to blend WordForm and TinyWordForm. If you have any suggestions...

**Comment by nijel** [comment]
Probably the function would have to be split to smaller separate ones (see handle_*). Or handle word addition in separate view and just redirect...

File: weblate/trans/forms.py [hunk 2]
```diff
@@ -589,6 +591,22 @@ class WordForm(forms.Form):
...
+    def __init__(self, *args, **kwargs):
+        super(TinyWordForm, self).__init__(*args, **kwargs)
+        self.helper = FormHelper()
+        self.helper.form_style = 'inline'
+        self.helper.form_show_labels = False
+        self.helper.form_id = 'addword'
+        self.helper.field_template = 'bootstrap3/layout/inline_field.html'
+        self.helper.layout = Layout(
+            PrependedText('source',_('Source')),
+            PrependedText('target',_('Translation')),
```
Comment (nijel) [inline weblate/trans/forms.py]: Codacy Issue found: [Exactly one space required after comma](https://www.codacy.com/app/michal_2/weblate/file/726968586/issues/source?bid=2038971&fileBranchId=2411671#l607)

File: weblate/trans/views/edit.py (hunk 1 above)
Comment (nijel) [inline weblate/trans/views/edit.py]: Codacy Issue found: [Using variable 'this_unit_url' before assignment](https://www.codacy.com/app/michal_2/weblate/file/726968729/issues/source?bid=2038971&fileBranchId=2411671#l484)

File: weblate/trans/forms.py [hunk 3]
```diff
@@ -589,6 +591,22 @@ class WordForm(forms.Form):
...
+    '''
+    def __init__(self, *args, **kwargs):
+        super(TinyWordForm, self).__init__(*args, **kwargs)
+        self.helper = FormHelper()
+        self.helper.form_style = 'inline'
+        self.helper.form_show_labels = False
+        self.helper.form_id = 'addword'
+        self.helper.field_template = 'bootstrap3/layout/inline_field.html'
+        self.helper.layout = Layout(
+            PrependedText('source',_('Source')),
```
Comment (nijel) [inline weblate/trans/forms.py]: Codacy Issue found: [Exactly one space required after comma](https://www.codacy.com/app/michal_2/weblate/file/726968586/issues/source?bid=2038971&fileBranchId=2411671#l606)

**Comment by nijel** [comment]
Thanks for your help, I've implemented it slightly differently in dbaa819 based on your code.

**Comment by emmapeel2** [comment]
Thank you very much!

Of course much more elegant solution than mine...

Nothing like bad code to put a dev to work on what you want :)

**Pull request closed without merging**